Lib/warnings.py

```diff
@@ -137,8 +137,16 @@ def filterwarnings(action, message="", category=Warning, module="", lineno=0,
     assert isinstance(module, str), "module must be a string"
     assert isinstance(lineno, int) and lineno >= 0, \
            "lineno must be an int >= 0"
-    _add_filter(action, re.compile(message, re.I), category,
-            re.compile(module), lineno, append=append)
+
+    if message:
+        regex = re.compile(message, re.I)
```

Comment:
This could be written shorter.
```python
regex = re.compile(message, re.I) if message else None
module = re.compile(module) if module else None
_add_filter(action, regex, category, module, lineno, append=append)
```